torchtext/prototype/t5/modules.py

```diff
@@ -0,0 +1,819 @@
+        )
+
+        x = attn[0]
+        if self.compute_relative_attention_bias and position_bias is None:
+            position_bias = attn[2]
+
+        return self.dropout1(x), position_bias
+
+    # feed forward block
+    def _ff_block(self, x: Tensor) -> Tensor:
+        x = self.linear2(self.dropout(self.activation(self.linear1(x))))
+        return self.dropout2(x)
+
+
+class T5Encoder(nn.TransformerEncoder):
+
+    r"""TransformerEncoder is a stack of N encoder layers. Users can build the
+    BERT(https://arxiv.org/abs/1810.04805) model with corresponding parameters.
+    Args:
+        encoder_layer: an instance of the TransformerEncoderLayer() class (required).
+        num_layers: the number of sub-encoder-layers in the encoder (required).
+        norm: the layer normalization component (optional).
+        enable_nested_tensor: if True, input will automatically convert to nested tensor
+            (and convert back on output). This will improve the overall performance of
+            TransformerEncoder when padding rate is high. Default: ``True`` (enabled).
+    Examples::
+        >>> encoder_layer = T5EncoderLayer(d_model=768, nhead=12, dim_feedfoward=3072, dropout=0.1, activation='relu', batch_first=True)
+        >>> t5_norm = T5LayerNorm(d_model=768)
+        >>> t5_encoder = T5Encoder(encoder_layer, num_layers=12, norm=t5_norm)
+        >>> src = torch.rand(10, 32, 512)
+        >>> out = t5_encoder(src)
+    """
+
+    def __init__(
+        self,
+        encoder_layer,
+        relative_attention_num_buckets,
+        num_heads,
+        num_layers=12,
+        norm=None,
+        enable_nested_tensor=True,
+    ):
+        super(T5Encoder, self).__init__(encoder_layer, num_layers, norm, enable_nested_tensor)
+
+        first_layer = copy.deepcopy(encoder_layer)
+        first_layer.compute_relative_attention_bias = True
+        first_layer.relative_attention_bias = nn.Embedding(relative_attention_num_buckets, num_heads)
+        self.layers = nn.ModuleList([first_layer] + [copy.deepcopy(encoder_layer) for i in range(num_layers - 1)])
+        self.num_layers = num_layers
+        self.enable_nested_tensor = enable_nested_tensor
+
+    def forward(
+        self, src: Tensor, mask: Optional[Tensor] = None, src_key_padding_mask: Optional[Tensor] = None
+    ) -> Tensor:
+        r"""Pass the input through the encoder layers in turn.
+        Args:
+            src: the sequence to the encoder (required).
+            mask: the mask for the src sequence (optional).
+            src_key_padding_mask: the mask for the src keys per batch (optional).
+        Shape:
+            see the docs in Transformer class.
+        """
+
+        output = src
+        position_bias = None
+        for mod in self.layers:
+            output, position_bias = mod(
+                output, src_mask=mask, src_key_padding_mask=src_key_padding_mask, position_bias=position_bias
+            )
+
+        return output
+
+
```

Comment:
`T5EncoderModel` is the complete encoder model that includes the word embeddings, and the final layer normalizations and dropouts